### PR TITLE
E2E: revise Invite: Revoke spec to clean up after itself.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Proposed Changes

This PR is primarily aimed at ensuring the `Invite: Revoke` spec cleans up after itself, but also contains a major reworking of the coverage the spec provides.

Key changes:
1. use the RestAPIClient to set up the test invite. 
2. add `afterAll` hook to handle potential clean-up of the invite should the test fail.
3. add invite-related calls to the RestAPIClient.

Details:
Recently, our approach to end-to-end testing has been to trim each spec to cover *just* the screens, flows and features being tested. This means that where possible, the preferred strategy is to delegate the setup and/or teardown processes to API-based calls. The idea here is to avoid testing flows and features unrelated to the intended test at hand and the flakiness that comes with it. 

In the context of `Invite: Revoke` spec, this means that we are only interested in the actual revocation of the invite portion, not the actual process to set up the invite itself. **As a result the steps to set up the invite has been removed from this spec**.

#### Testing Instructions

Ensure the following:
  - [x] Calypso E2E (desktop)
  - [x] Calypso E2E (mobile)

#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in [Simple](P9HQHe-k8-p2-p2), [Atomic](P9HQHe-jW-p2-p2), and [self-hosted Jetpack sites](PCYsg-g6b-p2-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Are we memoizing when appropriate (for expensive computations)? More info in [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md) and [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors)
- [ ] Have we sent any new strings [for translation](PCYsg-1vr-p2-p2) ASAP?

Related to #63714

### CODE REVIEW

File: packages/calypso-e2e/src/lib/pages/people-page.ts [hunk 1]
```diff
@@ -163,6 +164,8 @@ export class PeoplePage {
 		await this.waitUntilLoaded();
 
 		await this.page.click( selectors.revokeInviteButton );
-		await this.page.waitForSelector( selectors.inviteRevokedMessage );
+
+		const noticeComponent = new NoticeComponent( this.page );
+		await noticeComponent.noticeShown( 'Invite deleted' );
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/people-page.ts:169]: This change is not strictly related to the topic of this PR, but this is a good opportunity to update code that pre-date the new components.

File: packages/calypso-e2e/src/rest-api-client.ts [hunk 2]
```diff
@@ -22,7 +26,8 @@ import type { BodyInit, HeadersInit, RequestInit } from 'node-fetch';
 /**
  * Specifies the version of WordPress.com REST API.
  */
-type EndpointVersions = '1' | '1.1' | '1.2' | '1.3';
+type EndpointVersions = '1' | '1.1' | '1.2' | '1.3' | '2';
+type EndpointNamespace = 'rest' | 'wpcom';
```
Comment (worldomonation) [inline packages/calypso-e2e/src/rest-api-client.ts:30]: The [WordPress.com Developer Console](https://developer.wordpress.com/docs/api/console/) only supports versions 1 through 1.3 on the front end, but hidden away in the backend code is also version 2.

Combined with the `wpcom` namespace we are able to query undocumented endpoint that can perform tasks that the `rest` endpoint cannot.

File: packages/calypso-e2e/src/rest-api-client.ts [hunk 3]
```diff
@@ -127,10 +132,15 @@ export class RestAPIClient {
...
 	 * @param {string} endpoint REST API path.
+	 * @param {EndpointNamespace} [namespace] REST API namespace.
 	 * @returns {URL} Full URL to the endpoint.
 	 */
-	getRequestURL( version: EndpointVersions, endpoint: string ): URL {
-		const path = `/rest/v${ version }/${ endpoint }`.replace( /([^:]\/)\/+/g, '$1' );
+	getRequestURL(
+		version: EndpointVersions,
+		endpoint: string,
+		namespace: EndpointNamespace = 'rest'
```
Comment (worldomonation) [inline packages/calypso-e2e/src/rest-api-client.ts:141]: By default, most requests that we need to work with are in the `rest` namespace except to delete invites from the site.

File: packages/calypso-e2e/src/secrets/secrets-manager.ts [hunk 4]
```diff
@@ -151,7 +151,12 @@ export class SecretsManager {
 				totpUserInboxId: 'FAKE_VALUE',
 			},
 			testAccounts: {
-				defaultUser: { ...fakeAccount },
+				defaultUser: {
+					...fakeAccount,
+					userID: 0,
+					testSites: { primary: { id: 0, url: 'FAKE_VALUE' } },
+					email: 'FAKE_VALUE',
+				},
```
Comment (worldomonation) [inline packages/calypso-e2e/src/secrets/secrets-manager.ts:159]: `defaultUser` has had the `userID` or `email` attributes for some time, but it was not reflected in the type-checker here.

File: packages/calypso-e2e/src/rest-api-client.ts (hunk 2 above)
Comment (youbrokesomething) [reply packages/calypso-e2e/src/rest-api-client.ts:30]: Whoa. Thanks for this comment. This is really good to know.

**Review by youbrokesomething — APPROVED** [review]
LGTM
